core-plugins/src/main/java/io/cdap/plugin/batch/joiner/Joiner.java

```diff
@@ -291,7 +300,6 @@ void validateJoinKeySchemas(Map<String, Schema> inputSchemas, Map<String, List<S
   }
 
   Schema getOutputSchema(Map<String, Schema> inputSchemas, FailureCollector collector) {
-    validateJoinKeySchemas(inputSchemas, conf.getPerStageJoinKeys(), collector);
```

Comment:
why do we need to remove this? we still need this in case inputschemas are non-null right?

Reply:
`getOutputSchema()` is called from a couple of places:
1. `configurePipeline()`
2. `initialize()`
In both of these functions, `init()`, which calls `validateJoinKeySchemas()`,  is called before `getOuputSchema()`.